packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5499,6 +5519,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
     }
     assert(index >= 0, 'There are no routes below the specified anchorRoute.');
     _history[index].remove();
+    _history[index].route.didComplete(result);
```

Comment:
Perhaps we should incorporate the `didComplete` call into the `remove` method, but I'm happy to save that for a future PR :)

Reply:
I change .remove() to .complete(), work same and i think it's much cleaner like that.